src/pages/KakaoRedirect.tsx

```diff
@@ -0,0 +1,46 @@
+import { useNavigate } from 'react-router'
+import { useQuery } from '@tanstack/react-query'
+import { useEffect } from 'react'
+import axiosInstance from '../api/axiosInstance'
+
+const fetchLogin = async (code: string | null): Promise<string> => {
+  const { data } = await axiosInstance.get(`/api/oauth/login`, {
+    params: {
+      code,
+    },
+  })
+  return data
+}
+
+const KakaoRedirect = () => {
+  const navigate = useNavigate()
+  const code = new URL(document.location.toString()).searchParams.get('code')
```

Comment:
```suggestion
  const [searchParams] = useSearchParams()
  const code = searchParams.get('code')
```

react-router-dom에서 useSearchParams 훅을 제공하고 있어서 이를 활용할 수 있어요.